ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadTask.java

```diff
@@ -739,6 +741,23 @@ private int executeIncrementalLoad(long loadStartTime) throws Exception {
         LOG.info("Dropping table {} for optimised bootstarap", work.dbNameToLoadIn + "." + table);
         db.dropTable(work.dbNameToLoadIn + "." + table, true);
       }
+      Database sourceDb = getSourceDbMetadata();  //This sourceDb was the actual target prior to failover.
+      Map<String, String> sourceDbProps = sourceDb.getParameters();
+      List<String> replicationProps = MetaStoreUtils.getReplicationDbProps(); //Replication Props will be handled separately as part of preAckTask.
+      for (Map.Entry<String, String> currProp : targetDb.getParameters().entrySet()) {
+        String actualVal = sourceDbProps.get(currProp.getKey());
+        if (!replicationProps.contains(currProp.getKey()) && !currProp.getValue().equals(actualVal)) {
```

Comment:
can't we filter the map (targetDb.getParameters()) with replicationProps first to avoid the lookups?
````
    replicationProps.stream()
              .map(targetDb.getParameters()::get)
              .filter(Objects::isNull)
               ....
````